components/recent_date.tsx

```diff
@@ -14,37 +15,70 @@ type Props = {
     value: number | Date;
     children?(val: string): React.ReactElement | null;
     intl: IntlShape;
+    useTitleCase?: boolean;
+    dateTimeFormat?: Intl.DateTimeFormatOptions;
 }
 
 class RecentDate extends React.PureComponent<Props> {
     public render() {
-        const {value, timeZone} = this.props;
+        const {
+            value,
+            timeZone,
+            useTitleCase = true,
+            dateTimeFormat,
+            intl
+        } = this.props;
         const date = new Date(value);
 
-        if (isToday(date)) {
-            return (
-                <FormattedMessage
-                    id='date_separator.today'
-                    defaultMessage='Today'
-                />
-            );
-        } else if (isYesterday(date)) {
-            return (
-                <FormattedMessage
-                    id='date_separator.yesterday'
-                    defaultMessage='Yesterday'
-                />
-            );
+        const {is} = fromNow(date);
+
+        if (dateTimeFormat == null) {
+            if (useTitleCase && is.today) {
+                return (
+                    <FormattedMessage
+                        id='date_separator.today'
+                        defaultMessage='Today'
+                    />
+                );
+            } else if (useTitleCase && is.yesterday) {
+                return (
+                    <FormattedMessage
+                        id='date_separator.yesterday'
+                        defaultMessage='Yesterday'
+                    />
+                );
+            } else if (!useTitleCase && is.today) {
+                return (
+                    <FormattedRelativeTime
+                        value={0}
+                        unit='day'
+                        numeric='auto'
+                    />
+                );
+            } else if (!useTitleCase && is.yesterday) {
+                return (
+                    <FormattedRelativeTime
+                        value={-1}
+                        unit='day'
+                        numeric='auto'
+                    />
+                );
+            }
         }
 
-        const options = {
-            timeZone,
-            weekday: 'short',
-            month: 'short',
+        const options: Intl.DateTimeFormatOptions = dateTimeFormat ?? is.withinPastSevenDays ? {
```

Comment:
Could you maybe wrap the first part of this in brackets? Mixing `??` and a ternary statement is a bit difficult to read.

```suggestion
        const options: Intl.DateTimeFormatOptions = (dateTimeFormat ?? is.withinPastSevenDays) ? {
```

Reply:
Thanks for catching that @hmhealey... I clarified it and added a test for the that condition.